### PR TITLE
Stats: Extended number component with default formatter

### PR DESCRIPTION
**Pull request by grzegorz-cp**
#### Proposed Changes

* adding a default number formatter to Calypso components
* importing the formatter in Stats components and replacing it in places where it was used previously

#### Testing Instructions

* no visual changes should be present
* go to Stats page and check that the numbers on the page look correctly for your region
* Examples of affected places: components with the horizontal bar, All-time highlights section on the Insights page

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #71639

### CODE REVIEW

**Review by a8ck3n — APPROVED** [review]
Tests well for me.

File: packages/components/src/shortened-number/formattedNumber.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const FORMATTER = new Intl.NumberFormat();
+function formatNumber( number: number | null ) {
+	return Number.isFinite( number ) ? FORMATTER.format( number as number ) : '-';
+}
+
+const formatedNumber = ( number: number | null ) => formatNumber( number );
```
Comment (dognose24) [inline packages/components/src/shortened-number/formattedNumber.tsx]: Just nitpicking, is this variable supposed to be `formattedNumber`? 😄

**Review by dognose24 — COMMENTED** [review]
I left a message, which is not a blocker.

File: packages/components/src/shortened-number/stories/index.stories.jsx [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import ShortenedNumber from '..';
+import formttedNumber from './../formattedNumber';
```
Comment (jsnmoon) [inline packages/components/src/shortened-number/stories/index.stories.jsx]:
```suggestion
import formattedNumber from '../formattedNumber';
```

Fixes a typo and an unnecessarily verbose relative directory.

File: packages/components/src/shortened-number/formattedNumber.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const FORMATTER = new Intl.NumberFormat();
+function formatNumber( number: number | null ) {
```
Comment (jsnmoon) [inline packages/components/src/shortened-number/formattedNumber.tsx]:
```suggestion
export default function formattedNumber( number: number | null ) {
```

Could we directly export this function instead?

File: packages/components/src/shortened-number/formattedNumber.tsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const FORMATTER = new Intl.NumberFormat();
```
Comment (jsnmoon) [inline packages/components/src/shortened-number/formattedNumber.tsx]: This file should be dash-cased (`formatted-number.tsx`).

Comment (jsnmoon) [inline packages/components/src/shortened-number/formattedNumber.tsx]: Perhaps we could rename this folder to be `number-formatters`?

**Review by jsnmoon — CHANGES_REQUESTED** [review]

**Comment by grzegorz-cp** [comment]
Thanks team, I applied the changes.

**Review by jsnmoon — APPROVED** [review]
Let's ship this! Thanks for indulging my nitty feedback :)

**Review by dognose24 — APPROVED** [review]
LGTM!

**Pull request merged**